src/app/components/timeline/CardDialog.js

```diff
@@ -57,9 +70,62 @@ class CardDialog extends Component {
     this.props.closeDialog()
   }
 
+  addInputRef = (ref) => {
+    // Set state needs to be queued so that React does not overwrite
+    // the attribute in a merge
+    setTimeout(() => {
+      this.setState({
+        inputRefs: [...this.state.inputRefs, ref],
+      })
+    }, 0)
+  }
+
+  findChildInput = (id) => {
+    const foundInput = this.state.inputRefs.find((ref) => {
+      return ref && ref.props.id === id
+    })
+    return foundInput && findDOMNode(foundInput)
+  }
+
+  handleParagraphAttrChange = (attrName, desc) => {
+    this.setState({
+      paragraphs: {
+        ...this.state.paragraphs,
+        [attrName]: desc,
+      },
+    })
+  }
+
   saveEdit = () => {
     var newTitle = findDOMNode(this.refs.titleInput).value
-    this.props.actions.editCard(this.props.card.id, newTitle, this.state.description)
+    const attrs = {}
+    this.props.customAttributes.forEach(attr => {
+      const { name, type } = attr
+      if (type == 'paragraph') {
+        attrs[name] = this.state.paragraphs[name] || this.props.card[name]
+      } else {
+        const val = this.findChildInput(`${name}Input`).value
+        attrs[name] = val
+      }
+    })
```

Comment:
This looks for values for all of the custom attributes.  If a child is unmounted then the ref isn't valid.
(BTW string-refs are also deprecated)

Reply:
re: string-refs

that'll be a refactor that touches just about everything, but we should stop using those